pkg/http/execute_job_handler.go

```diff
@@ -74,11 +83,18 @@ func createJobExecution(name, ns string, body io.ReadCloser) *v1alpha1.JobExecut
 	return &v1alpha1.JobExecution{
 		ObjectMeta: metav1.ObjectMeta{
 			GenerateName: name + "-",
-			Namespace:    ns,
+			Namespace:    namespace,
+			Labels:       jobTemplate.ObjectMeta.Labels,
 		},
 		Spec: v1alpha1.JobExecutionSpec{
 			JobTemplateName: name,
 			Payload:         b.String(),
 		},
 	}
 }
+
+func (e *executeJobHandler) getJobTemplate(namespace, name string, ctx context.Context) (*v1alpha1.JobTemplate, error) {
+	jt := new(v1alpha1.JobTemplate)
+	err := e.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, jt)
```

Comment:
More of a learning question for me. I'm lost how `e` has a `Get` method. Is it some kind of inheritance from the *Server type in the struct? I know go doesn't have inheritance, so this is a mystery to me.

Reply:
It's called embedding structures/embedding anonymous structures (or you can even find it if you search inheritance in Go, although is more like composition): https://golangbyexample.com/embedding-interfaces-go/

Reply:
Thank you! Now I understand this `Get` method comes from a nested k8s `client.Client`. This embedding of anonymous structures looks powerful but can be confusing at times i.e. it's not clear which embedded struct/interface contains the method being invoked. And what of clashes in embedded structs/interface method names? I guess those should be compile time errors.

Just thinking out loud, and appreciating the learning!

Reply:
Yes, I think you'll get a compilation error. However, you can still call `client.Client`'s `Get` function by doing `e.Client.Get(...)`